### PR TITLE
Add new metric - purefb_buckets_quota_space_bytes , purefb_buckets_object_count

### PR DESCRIPTION
**Pull request by chrroberts-pure**
In issue #41 I was investigating bucket metrics - and found it would be very easy to implement quota metrics.

Here is the suggested quota space bytes - tested on 3.3.x (doesn't support quotas - will show false and 0 for all buckets) and 4.1.x - below is output from a 4.1.x FB with all 3 types of configurations

````
# HELP purefb_buckets_quota_space_bytes FlashBlade buckets quota space in bytes
# TYPE purefb_buckets_quota_space_bytes gauge
purefb_buckets_quota_space_bytes{hard_limit_enabled="false",name="bucket1"} 0
purefb_buckets_quota_space_bytes{hard_limit_enabled="true",name="bucket2"} 2.199023255552e+12
purefb_buckets_quota_space_bytes{hard_limit_enabled="false",name="bucket3"} 2.199023255552e+12
````
````
# HELP purefb_buckets_object_count FlashBlade buckets object count
# TYPE purefb_buckets_object_count gauge
purefb_buckets_object_count{name="bucket1"} 26211
purefb_buckets_object_count{name="bucket2"} 0
purefb_buckets_object_count{name="bucket3"} 2.111109e+06
````

It was decided by the practice to keep `purefb_buckets_space_bytes{name="<bucket>",space="object_count"}` for now - to be depreciated in v1.0.10 - release notes to be included. 

--

The metric `purefb_hardware_connectors_performance_errors` was misspelled as `purefb_shardware_connectors_performance_errors` - please make a note in the release notes, I've corrected it in this release. 

documentation is in #43

### CODE REVIEW

**Review by sdodsley — APPROVED** [review]

**Review by sdodsley — APPROVED** [review]

**Review by genegr — DISMISSED** [review]

**Comment by chrroberts-pure** [comment]
I have also changed the api datatypes for `ArraySpace`, `ArraySpaceList`, `EradicationConfig`, `Bucket`, `BucketsList`, `Space` , on our rest-client to align with the models from the PurityAPI and converting them when collected as I was having issues with some conversations. Happy to chat about this change - there's probably a decision to be made here, but my gut is to have the `rest-client` match the PurityAPI model.

File: internal/openmetrics-exporter/common_tests.go [hunk 1]
```diff
@@ -25,24 +24,3 @@ func metricsCheck(t *testing.T, c prometheus.Collector, want map[string]bool) {
...
-		metrics[buff.String()] = true
-	}
-        for a := range metrics {
-		fmt.Println(a)
-	}
-	fmt.Println("++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++")
-        for a := range want {
-		fmt.Println(a)
-	}
-}
```
Comment (chrroberts-pure) [inline internal/openmetrics-exporter/common_tests.go:48]: @genegr there was also this function that was not being used, please confirm its removal.

**Review by james-laing — DISMISSED** [review]
LGTM

**Comment by chrroberts-pure** [comment]
please do not merge yet - there's a package we use that has been deprecated - looking deeper into its impact without its use. https://matttproud.com/blog/posts/retiring-pbutil.html

**Comment by chrroberts-pure** [comment]
I have confirmed that the `golang_protobuf_extensions` pakcage is no longer a dependency after the upgrade of `prometheus/common` package. I've tested this with a production FlashBlade with both the binary and the docker-build container.

**Review by sdodsley — APPROVED** [review]

**Pull request merged**